portfolio/src/main/webapp/script.js

```diff
@@ -58,23 +58,46 @@ function getRandomName() {
 }
 
 /**
- * Fetches comments from DataServlet and adds them to the DOM as a list.
+ * Fetches the last commentsLimit comments from DataServlet and adds them to the DOM as a list.
+ * commentsLimit is selected by the user and sent to the server as parameter in the query string.
  */
 function getComments() {
-  fetch('/data').then(response => response.json()).then((comments) => {
+  const dataURL = `/data?commentsLimit=${document.getElementById('commentsLimit').value}`;
 
-    const commentsListElement = document.getElementById('comments-history');
+  fetch(dataURL).then(response => {
+      if (response.status == 200) {
+        return response.json();
+      } else {
+        if (response.status == 400) {
+          throw new Error("Invalid data: The number of comments should be between 0 and 100");
```

Comment:
Good stuff. Another timer we can talk about taking this a bit further (e.g. sending back an error message from the server as a JSON payload).